app/helpers/spree/frontend_helper_decorator.rb

```diff
@@ -1,50 +1,70 @@
+Spree::FrontendHelper.class_eval do
 
-    def spree_breadcrumbs(taxon, _separator = '', product = nil)
-        return ''  if current_page?('/') || taxon.nil?
+  def taxons_tree(root_taxon, current_taxon, max_level = 3)
 
-        # breadcrumbs for root
-        crumbs = [content_tag(:li, content_tag(
-          :a, content_tag(
-            :span, Spree.t(:home), itemprop: 'name'
-          ) << content_tag(:meta, nil, itemprop: 'position', content: '0'), itemprop: 'url', href: spree.root_path
-        ) << content_tag(:span, nil, itemprop: 'item', itemscope: 'itemscope', itemtype: 'https://schema.org/Thing', itemid: spree.root_path), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item')]
-  
-        if taxon
-          ancestors = taxon.ancestors.where.not(parent_id: nil)
-  
-          # breadcrumbs for ancestor taxons
-          crumbs << ancestors.each_with_index.map do |ancestor, index|
-            content_tag(:li, content_tag(
-              :a, content_tag(
-                :span, ancestor.name, itemprop: 'name'
-              ) << content_tag(:meta, nil, itemprop: 'position', content: index + 1), itemprop: 'url', href: seo_url(ancestor, params: permitted_product_params)
-            ) << content_tag(:span, nil, itemprop: 'item', itemscope: 'itemscope', itemtype: 'https://schema.org/Thing', itemid: seo_url(ancestor, params: permitted_product_params)), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item')
-          end
-  
-          # breadcrumbs for current taxon
-          crumbs << content_tag(:li, content_tag(
-              :div,content_tag(
-              :span, taxon.name, itemprop: 'name'
-            ) << content_tag(:meta, nil, itemprop: 'position', content: ancestors.size + 1), itemprop: 'url', href: seo_url(taxon, params: permitted_product_params)
-          ) << content_tag(:span, nil, itemprop: 'item', itemscope: 'itemscope', itemtype: 'https://schema.org/Thing', itemid: seo_url(taxon, params: permitted_product_params)), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item active' )
-  
-          # breadcrumbs for product
-          if product
-            crumbs << content_tag(:li, content_tag(
-              :span, content_tag(
-                :span, product.name, itemprop: 'name'
-              ) << content_tag(:meta, nil, itemprop: 'position', content: ancestors.size + 2), itemprop: 'url', href: spree.product_path(product, taxon_id: taxon&.id)
-            ) << content_tag(:span, nil, itemprop: 'item', itemscope: 'itemscope', itemtype: 'https://schema.org/Thing', itemid: spree.product_path(product, taxon_id: taxon&.id)), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item ')
+    return '' if max_level < 1
+    selected_parent_taxon_name = params["id"]
```

Comment:
variable name expecting name not id. Also why we need id of taxon here. Please justify.?

Reply:
in id we are getting category/name that is why we are using it

Reply:
Thats making sense. Thanks